modules/24_trade/se_trade/datainput.gms

```diff
@@ -180,5 +180,70 @@ $ifthen.import_h2_EU "%cm_Ger_Pol%" == "ensec"
   p24_seTradeCapacity(t,regi,regi2,entySe)$(t.val eq 2030) = p24_seTrade_Quantity(regi,regi2,entySe)*0.2;
 $endif.import_h2_EU
 
+$ifthen.import_nzero_EU "%cm_import_EU%" == "nzero"
+
+*** H2 trade:
+***   Importing regions: Germany, 100 TWh/yr, and EWN, proportional to German values and FE|Gases demand by 2050 in the reference NPi run.
+***   Exporting regions: UK, Norway and Spain (one-third each)
+***   exponential curve starting at 12 TWh by 2030 for Germany
+*** 100 TWh = 100 * 0.0036 EJ/yr =  100 * 0.0036 * sm_EJ_2_TWa TWa = 100 * 0.0036 * 0.03171 TWa = 0.0114156 TWa
+
+*** calculating the share of FE demand per carrier for each region in each region group
+  p24_seAggReference(ttot,regi,seAgg) = sum(enty$seAgg2se(seAgg,enty), sum(se2fe(enty,enty2,te), pm_prodFEReference(ttot,regi,enty,enty2,te)));
+  p24_FEShareInRegion(ttot,ext_regi,regi,seAgg)$(regi_group(ext_regi,regi) and p24_seAggReference(ttot,regi,seAgg)) = p24_seAggReference(ttot,regi,seAgg) / sum(regi2$regi_group(ext_regi,regi2), p24_seAggReference(ttot,regi2,seAgg));
+
+*** defining Germany H2 trade import flows
+*** 2050 and onward
+  p24_seTradeCapacity(t,"UKI","DEU","seh2")$(t.val ge 2050) = 100 * 0.0036 * sm_EJ_2_TWa * 1/3;
+  p24_seTradeCapacity(t,"NEN","DEU","seh2")$(t.val ge 2050) = 100 * 0.0036 * sm_EJ_2_TWa * 2/3;
+*** 2030
+  p24_seTradeCapacity("2030","UKI","DEU","seh2") = 12 * 0.0036 * sm_EJ_2_TWa * 1/3;
+  p24_seTradeCapacity("2030","NEN","DEU","seh2") = 12 * 0.0036 * sm_EJ_2_TWa * 2/3;
+
+*** defining EWN H2 trade import flows
+*** 2050 and onward
+  p24_seTradeCapacity(t,"UKI","EWN","seh2")$(t.val ge 2050) = 100 * 0.0036 * sm_EJ_2_TWa * (p24_FEShareInRegion("2050","EUR_regi","EWN","all_sega")/p24_FEShareInRegion("2050","EUR_regi","DEU","all_sega")) * 1/3;
+  p24_seTradeCapacity(t,"ESW","EWN","seh2")$(t.val ge 2050) = 100 * 0.0036 * sm_EJ_2_TWa * (p24_FEShareInRegion("2050","EUR_regi","EWN","all_sega")/p24_FEShareInRegion("2050","EUR_regi","DEU","all_sega")) * 2/3;
+*** 2030
+  p24_seTradeCapacity("2030","UKI","EWN","seh2") = 12 * 0.0036 * sm_EJ_2_TWa * (p24_FEShareInRegion("2050","EUR_regi","EWN","all_sega")/p24_FEShareInRegion("2050","EUR_regi","DEU","all_sega")) * 1/3;
```

Comment:
I would find 
`p24_seTradeCapacity("2030",regi,regi2,"seh2") = 12/100 * p24_seTradeCapacity("2050",regi,regi2,"seh2")` 

easier to read :-)

Reply:
I will keep as is, as if we extend the assumptions in the future to include more regional trade flows, the current formulation is more descriptive of the changes that need to be made, instead of hard-codding numbers.

Reply:
fine with me, I was just voicing my personal preference :-)